File: 189-RotateArray/javascript/solution.js
// Last updated: 3/6/2026, 12:45:53 AM
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
    k = k % nums.length
    function reverse(i, j) {
        while (i < j) {
            [nums[i], nums[j]] = [nums[j], nums[i]]
            i++
            j--
        }
    }
    reverse(0, nums.length - 1)
    reverse(0, k - 1)
    reverse(k, nums.length - 1)
    return nums;
}

File: 189-RotateArray/cpp/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k = k % nums.size();
        auto reverse=[&](int i,int j) {
            while (i < j) {
                swap(nums[i], nums[j]);
                i++;
                j--;
            }
        };
        reverse(0, nums.size() - 1);
        reverse(0, k - 1);
        reverse(k, nums.size() - 1);
    }
};